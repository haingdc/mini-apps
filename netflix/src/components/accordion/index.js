import React , { createContext, useState } from 'react';

var ToggleContext = createContext();

export default function Accordion({ children, ...rest }) {
  return (
    <Container {...rest}>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function AccordionTitle({ children, ...rest }) {
  return <Title { ...rest }>{ children }</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...rest }) {
  return <Frame { ...rest }>{ children }</Frame>
}

Accordion.Item = function AccordionItem({ children, ...rest }) {
  var [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow}}>
      <Item {...rest}>{ children }</Item>
    </ToggleContext.Provider>
  );
}

Accordion.Header = function AccordionHeader({ children, ...rest }) {
  return <Header { ...rest } onClick{() => setToggle}>{ children }</Header>
}