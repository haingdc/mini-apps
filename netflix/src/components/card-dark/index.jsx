import './index.scss';
import { IoCheckmarkOutline } from 'react-icons/io5';

export default function CardDark() {
  return (
    <div className="card">
      <div className="card__label--main">Create an Event</div>
      <div className="card__body">
        <div className="card__row">
          <input type="text" className="card__title" placeholder="Title" value="Drink a capuchino" />
        </div>
        <div className="card__row">
          <div className="card__label card__row__label">Color</div>
          <div className="card__row__colors">
            <label className="container">
              <input name="color" type="radio" value="red" />
              <span className="checkmark">
                <IoCheckmarkOutline className="checkmark__ico" />
              </span>
            </label>
            <label className="container">
              <input name="color" type="radio" value="orange"/>
              <span className="checkmark">
                <IoCheckmarkOutline className="checkmark__ico" />
              </span>
            </label>
            <label className="container">
              <input name="color" type="radio" value="yellow"/>
              <span className="checkmark">
                <IoCheckmarkOutline className="checkmark__ico" />
              </span>
            </label>
            <label className="container">
              <input name="color" type="radio" value="green"/>
              <span className="checkmark">
                <IoCheckmarkOutline className="checkmark__ico" />
              </span>
            </label>
            <label className="container">
              <input name="color" type="radio" value="blue"/>
              <span className="checkmark">
                <IoCheckmarkOutline className="checkmark__ico" />
              </span>
            </label>
            <label className="container">
              <input name="color" type="radio" value="purple"/>
              <span className="checkmark">
                <IoCheckmarkOutline className="checkmark__ico" />
              </span>
            </label>
          </div>
        </div>
        <div className="card__row">
          <div className="card__label card__row__label">Date</div>
        </div>
        <div className="card__row">
        <div className="card__date">
            <input type="date"/>
          </div>
        </div>
        <div className="card__row card__row--time">
          <div className="card__time">
            <input type="time"/>
          </div>
          <div className="card__time">
            <input type="time"/>
          </div>
        </div>
      </div>
      <div className="card__footer">
        <div className="card__row">
          <div class="buttons-group">
            <button class="card__button card__button--cancel">Cancel</button>
            <button class="card__button">Create</button>
          </div>
        </div>
      </div>
    </div>
  );
}