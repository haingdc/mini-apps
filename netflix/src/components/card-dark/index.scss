:root {
  --color-white-1: #fff;
}
.card {
  background: #3d3c40;
  display: inline-flex;
  flex-direction: column;
  border-radius: 15px;
  font-size: 10px;
  line-height: 1;
  width: 328px;
  overflow: hidden;

  &__label {
    font-size: 16px;
    color: #9b98a6;
    &--main {
      color: white;
      font-size: 20px;
      line-height: 1.7;
    }
  }
  &__title {
    background-color: transparent;
    outline: none;
    border: none;
    border-bottom: 2px solid #4f4e52;
    width: 100%;
    color: hsl(252.9, 7.3%, 70%);
    caret-color: white;
    font-size: 20px;
    margin-bottom: 10px;
    padding: 9px 0;
    &::placeholder {
      color: hsl(252.9, 7.3%, 60%);
    }
  }
  &__date {
    --shadow: 0px 0.5px 0.5px rgba(255,255,255,0.1) inset, 2px 10px 16px rgba(0,0,0,0.2);
    flex: 1 1 auto;
    input {
      width: 100%;
      border-radius: 5px;
      border: none;
      height: 32px;
      background-color: #313033;
      color: #fff;
      padding-left: 10px;
      outline: none;
      box-shadow: var(--shadow);
      &:focus-visible {
        box-shadow: 0 0 0 1px #6562fc;
      }
    }
  }
  &__time {
    flex: 1 1 50%;
    --shadow: 0px 0.5px 0.5px rgba(255,255,255,0.1) inset, 2px 10px 16px rgba(0,0,0,0.2);
    input {
      width: 100%;
      border-radius: 5px;
      border: none;
      height: 32px;
      padding-left: 10px;
      background-color: #4d4b52;
      color: #fff;
      outline: none;
      box-shadow: var(--shadow);
      &:focus-visible {
        box-shadow: 0 0 0 1px #6562fc;
      }
    }
  }
  &__row {
    display: flex;
    margin-top: 11px;
    align-items: center;
    gap: 10px;
    &:first-child {
      margin-top: 0;
    }
    &__label {
      flex: 0 0 50px;
      margin-top: 4px;
    }
    &__colors {
      flex: 1 1 auto;
      input[name=color] {
        background-color: red;
      }
    }
    &--time {
      gap: 5px;
    }
  }
  &__header {
    display: flex;
    padding: 20px 20px 0;
    margin-bottom: 13px;
    justify-content: space-between;
    align-items: center;
  }
  &__body {
    display: flex;
    flex-direction: column;
    padding: 0 20px 30px;
    box-shadow: 0px -1px 0px rgba(0,0,0,0.3) inset;
  }
  &__footer {
    background-color: #313033;
    padding: 16px;
    > div {
      justify-content: flex-end;
    }
  }
  .buttons-group {
    display: flex;
    gap: 8px;
  }
  &__button {
    --background-color: #6562fc;
    --color: #fff;
    --shadow: 0px 0.5px 0px rgba(255,255,255,0.8) inset, 2px 2px 2px rgba(0,0,0,0.2);
    background-color: var(--background-color);
    color: var(--color);
    border: none;
    border-radius: 5px;
    font-size: 12px;
    padding: 10px 19px;
    font-weight: 500;
    box-shadow: var(--shadow);
    &--cancel {
      --background-color: #3e3d40;
      --color: #9b98a6;
      --shadow: 0px 0.5px 0.5px rgba(0,0,0,0.2) inset, 2px 2px 2px rgba(0,0,0,0.2);
    }
  }
}

.container {
  position: relative;
  padding-left: 28px;
  cursor: pointer;
  font-size: 22px;
  user-select: none;
  display: inline-block;
  width: 22px;
  height: 22px;
}

input[name=color] {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}



.checkmark {
  --radio-background: #eee;
  position: absolute;
  top: 0;
  left: 0;
  height: 22px;
  width: 22px;
  background-color: var(--radio-background);
  border-radius: 50%;

  display: flex;
  justify-content: center;
  align-items: center;
  .container:nth-child(1) & {
    --radio-background: #f9371c;
  }
  .container:nth-child(2) & {
    --radio-background: #f97c1c;
  }
  .container:nth-child(3) & {
    --radio-background: #f9c81c;
  }
  .container:nth-child(4) & {
    --radio-background: #41d0b6;
  }
  .container:nth-child(5) & {
    --radio-background: #2cadf6;
  }
  .container:nth-child(6) & {
    --radio-background: #6562fc;
  }
}

.container:hover input ~ .checkmark {
  // background-color: #ccc;
}

.container input:checked ~ .checkmark {
  // background-color: #2196F3;
}

.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

.container input:checked ~ .checkmark:after {
  display: block;
}

.container .checkmark:after {
  --border-color: hsl(206.6, 49.7%, 39.1%);
	border-radius: 50%;
  background: transparent;
  width: 25px;
  height: 25px;
  border: 1.8px solid var(--border-color);
}

.container:nth-child(1) .checkmark:after {
  --border-color: hsl(13.3, 100%, 33.13%)
;
}
.container:nth-child(2) .checkmark:after {
  --border-color: hsl(26.1, 64.8%, 40.3%);
}
.container:nth-child(3) .checkmark:after {
  --border-color: hsl(26.7, 34.8%, 44.3%);
}
.container:nth-child(4) .checkmark:after {
  --border-color: hsl(160, 30%, 40%);
}
.container:nth-child(5) .checkmark:after {
  --border-color: hsl(206.6, 49.7%, 39.1%);
}
.container:nth-child(6) .checkmark:after {
  --border-color: hsl(248, 40%, 46%);
}

.checkmark__ico {
  font-size: 15px;
  opacity: 0;
}

.container input:checked ~ .checkmark .checkmark__ico {
  opacity: 1;
}