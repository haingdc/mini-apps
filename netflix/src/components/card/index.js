import { Container, Item, Image, Feature, FeatureTitle, FeatureText, FeatureClose, Maturity, Content, Entities, Meta, Group, Text, Title, SubTitle } from './styles/card';
import React , { createContext, useContext, useState } from 'react';

export var FeatureContext = createContext();

export default function Card(props) {
  var { children, ...rest } = props;
  var [showFeature, setShowFeature] = useState(false);
  var [itemFeature, setItemFeature] = useState({});

  return (
    <FeatureContext.Provider
      value={{showFeature, setShowFeature, itemFeature, setItemFeature}}
    >
      <Container {...rest}>{children}</Container>
    </FeatureContext.Provider>
  );
}

Card.Text = function CardText({ children, ...rest }) {
  return <Text {...rest}>{children}</Text>
}

Card.Title = function CardTitle({ children, ...rest }) {
  return <Title {...rest}>{children}</Title>
}

Card.SubTitle = function CardSubTitle({ children, ...rest }) {
  return <SubTitle {...rest}>{children}</SubTitle>
}

Card.Group = function CardGroup({ children, ...rest }) {
  return <Group {...rest}>{children}</Group>
}

Card.Meta = function CardMeta({ children, ...rest }) {
  return <Meta {...rest}>{children}</Meta>
}

Card.Entities = function CardEntities({ children, ...rest }) {
  return <Entities {...rest}>{children}</Entities>
}

Card.Item = function CardItem(props) {
  var { children, item, ...rest } = props;
  var { setShowFeature, setItemFeature } = useContext(FeatureContext);
  return (
    <Item
      onClick={() => {
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...rest}
    >
      {children}
    </Item>
  );
}

Card.Image = function CardImage(props) {
  return <Image {...props} />;
}

Card.Feature = function CardFeature(props) {
  var { children, category, ...rest } = props;
  var { showFeature, itemFeature, setShowFeature } = useContext(FeatureContext);
  console.log({ showFeature })
  return showFeature ? (
    <Feature src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`} {...rest}>
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature(false)}>
          <img src="/images/icons/close.png" alt="Close" />
        </FeatureClose>
      </Content>

      <Group margin="30px 0" flexDirection="row" alignItems="center">
        <Maturity rating={itemFeature.maturity}>
          {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
        </Maturity>
        <FeatureText fontWeight="bold">
          {itemFeature.genre.charAt(0).toUpperCase() + itemFeature.genre.slice(1)}
        </FeatureText>
      </Group>
    </Feature>
  ) : null;
}