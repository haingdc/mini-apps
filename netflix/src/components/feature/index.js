import React from 'react';
import { Container, Title, SubTitle } from './styles/feature';

export default function Feature(props) {
  var { children, ...rest } = props;
  return <Container {...props}>{children}</Container>;
}

Feature.Title = function FeatureTitle(props) {
  var { children, ...rest } = props;
  return <Title {...rest}>{children}</Title>;
}

Feature.SubTitle = function SubFeatureTitle(props) {
  var { children, ...rest } = props;
  return <SubTitle {...rest}>{children}</SubTitle>;
}
