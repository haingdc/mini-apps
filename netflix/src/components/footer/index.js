import React from 'react';
import { Container , Row, Column, Link, Title, Text, Break } from './styles/footer';

export default function Footer({ children, ...rest }) {
  return <Container {...rest}>{ children }</Container>;
}

Footer.Row = function FooterRow({ children, ...rest }) {
  return <Row {...rest}>{ children }</Row>;
}

Footer.Column = function FooterRow({ children, ...rest }) {
  return <Column {...rest}>{ children }</Column>;
}

Footer.Link = function FooterRow({ children, ...rest }) {
  return <Link {...rest}>{ children }</Link>;
}

Footer.Title = function FooterRow({ children, ...rest }) {
  return <Title {...rest}>{ children }</Title>;
}

Footer.Text = function FooterRow({ children, ...rest }) {
  return <Text {...rest}>{ children }</Text>;
}

Footer.Break = function FooterRow({ children, ...rest }) {
  return <Break {...rest}>{ children }</Break>;
}


