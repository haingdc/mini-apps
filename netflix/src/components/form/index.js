import React from 'react';
import { Container, Error, Title, Text, TextSmall, Link, Input, Submit, Base } from './styles/form';

export default function Form(props) {
  var { children, rest } = props;
  return <Container {...rest}>{children}</Container>;
}

Form.Error = function FormError(props) {
  var { children, ...rest } = props;
  return <Error {...rest}>{children}</Error>;
}

Form.Base = function FormBase(props) {
  var { children, ...rest } = props;
  return <Base {...rest}>{children}</Base>;
}

Form.Input = function FormInput(props) {
  return <Input {...props} />;
}

Form.Title = function FormTitle(props) {
  var { children, ...rest } = props;
  return <Title {...rest}>{children}</Title>;
}

Form.Text = function FormText(props) {
  var { children, ...rest } = props;
  return <Text {...rest}>{children}</Text>;
}

Form.TextSmall = function FormTextSmall(props) {
  var { children, ...rest } = props;
  return <TextSmall {...rest}>{children}</TextSmall>;
}

Form.Link = function FormLink(props) {
  var { children, ...rest } = props;
  return <Link {...rest}>{children}</Link>;
}

Form.Submit = function FormSubmit(props) {
  var { children, ...rest } = props;
  return <Submit {...rest}>{children}</Submit>;
}