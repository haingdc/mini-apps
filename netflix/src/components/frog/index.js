import React, { createContext, useContext, useLayoutEffect, useState } from 'react';
import { Container } from './styles/frog'

export var FrogContext = createContext();

export default function Frog(props) {
  var { mode, animationStartTime } = useContext(FrogContext);
  var [left, setLeft] = useState(55);
  var [top, setTop] = useState(205);
  var [backgroundPosition, setBackgroundPosition] = useState(0);

  useLayoutEffect(() => {
    if (mode === 'play') {
      var timerId = window.requestAnimationFrame(animate);

      function animate(time) {
        var left = (55 + (time - animationStartTime)/10 % 300);
        var top = (205 - 10 * ((time - animationStartTime)/100 % 10) + ((time - animationStartTime)/100 % 10) * ((time - animationStartTime)/100 % 10) );
        var t = (time - animationStartTime)/10 % 100;
        var backgroundPosition = - Math.floor(t / (100/2)) * 60;
        setLeft( left );
        setTop( top );
        setBackgroundPosition( backgroundPosition );
        timerId = window.requestAnimationFrame(animate);
      }
    }

    return () => window.cancelAnimationFrame(timerId);
  }, [ mode ]);
  return <Container backgroundPosition={backgroundPosition} left={left} top={top} />;
}

Frog.Container = function FrogContainer({ children, ...rest }) {
  var [requestId, setRequestId] = useState(0);
  var [animationStartTime, setAnimationStartTime] = useState(0);
  var [mode, setMode] = useState('stop');
  return (
    <FrogContext.Provider
      value={{ mode, setMode, requestId, setRequestId, animationStartTime, setAnimationStartTime }}
    >
      {children}
    </FrogContext.Provider>
  );
}

Frog.StartButton = function FrogButton() {
  const { setMode, setAnimationStartTime } = useContext(FrogContext);
  function play() {
    setAnimationStartTime( window.performance.now() );
    setMode('play');
  }
  return <span onClick={play}>Start</span>;
}

Frog.PauseButton = function FrogButton() {
  const { setMode, requestId, setRequestId } = useContext(FrogContext);
  function stop() {
    if (requestId) {
      window.cancelAnimationFrame(requestId);
    }
    setRequestId(0);
    setMode('stop');
  }
  return <span onClick={stop}>Pause</span>;
}