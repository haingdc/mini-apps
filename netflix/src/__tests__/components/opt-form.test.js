import React from 'react';
import { render } from '@testing-library/react';
import { OptForm } from '../../components';

describe('<OptForm />', () => {
  it('renders the <OptForm /> with populated data', () => {
    var { container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Input placeholder="Email address" />
        <OptForm.Button>Try it now</OptForm.Button>
        <OptForm.Break />
        <OptForm.Text>Ready to watch? Enter your email to create or restart your membership.</OptForm.Text>
      </OptForm>
    );

    expect(getByText('Try it now')).toBeTruthy();
    expect(getByPlaceholderText('Email address')).toBeTruthy();
    expect(getByText('Ready to watch? Enter your email to create or restart your membership.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});