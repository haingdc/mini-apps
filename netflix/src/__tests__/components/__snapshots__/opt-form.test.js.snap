// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> renders the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-uz9nhu-0 gcnknG"
>
  <input
    class="opt-form__Input-uz9nhu-1 hfmQUa"
    placeholder="Email address"
  />
  <button
    class="opt-form__Button-uz9nhu-2 jSmuCg"
  >
    Try it now
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <div
    class="opt-form__Break-uz9nhu-4 OUzIE"
  />
  <h1
    class="opt-form__Text-uz9nhu-3 cvzEXu"
  >
    Ready to watch? Enter your email to create or restart your membership.
  </h1>
</div>
`;
